### DISCUSSION TITLE
Samples should dispose the clients?

### DISCUSSION THREAD

**Discussion by taspeotis** [kubernetes-client/csharp#864] [Ideas]
The readme and example folders are full of things like:

```
var client = new Kubernetes(config);
```

Surely this should be `using var client = ...` since `Kubernetes` implements `IDisposable`?

**Comment by tg123** [comment]
Nice to have, but not a must.
it is a good practice to call `Dispose` if the obj impled it.
The reason Kuberenetes object is Disposable is because the inner property httpclient is Disposable. The dispose will call httpclient's dispose to clean up all httpclient's resources.


To set a better example, you are welcome to send a PR to add `using` to them

**Comment by zhiweiv** [comment]
I think it is recommended to not dispose httpclient frequently,  should use it as singleton, so it is good to use singleton k8s client also.
https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests#issues-with-the-original-httpclient-class-available-in-net

> The original and well-known [HttpClient](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient) class can be easily used, but in some cases, it isn't being properly used by many developers.
> 
> Though this class implements IDisposable, declaring and instantiating it within a using statement is not preferred because when the HttpClient object gets disposed of, the underlying socket is not immediately released, which can lead to a socket exhaustion problem.
